File: 351003/Pulish/app/services/user_service.py
from app.dto.user import UserRequestTo, UserResponseTo
from app.models.user import User
from app.repository.inmemory import InMemoryRepository
from app.core.exceptions import NotFoundException


class UserService:
    def __init__(self):
        self.repo = InMemoryRepository()

    def create(self, dto: UserRequestTo) -> UserResponseTo:
        user = User(
            id=0,
            login=dto.login,
            password=dto.password,
            firstname=dto.firstname,
            lastname=dto.lastname
        )
        saved = self.repo.create(user)
        return self._to_response(saved)

    def find_all(self):
        return [self._to_response(u) for u in self.repo.find_all()]

    def find_by_id(self, id: int):
        user = self.repo.find_by_id(id)
        if not user:
            raise NotFoundException("User not found", 40401)
        return self._to_response(user)

    def update(self, id: int, dto: UserRequestTo):
        user = self.repo.find_by_id(id)
        if not user:
            raise NotFoundException("User not found", 40401)
        user.login = dto.login
        user.firstname = dto.firstname
        user.lastname = dto.lastname
        return self._to_response(user)

    def delete(self, id: int):
        user = self.repo.find_by_id(id)
        if not user:
            raise NotFoundException("User not found", 40401)
        self.repo.delete(id)

    def _to_response(self, user: User) -> UserResponseTo:
        return UserResponseTo(
            id=user.id,
            login=user.login,
            firstname=user.firstname,
            lastname=user.lastname
        )

File: 351004/Mukhin/Lab1/restApi/src/main/java/com/example/restApi/services/UserService.java
package com.example.restApi.services;

import com.example.restApi.dto.request.UserRequestTo;
import com.example.restApi.dto.response.UserResponseTo;
import com.example.restApi.exception.NotFoundException;
import com.example.restApi.mapper.UserMapper;
import com.example.restApi.model.User;
import com.example.restApi.repository.UserRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class UserService {
    private final UserRepository userRepository;
    private final UserMapper userMapper;

    public UserService(UserRepository userRepository, UserMapper userMapper) {
        this.userRepository = userRepository;
        this.userMapper = userMapper;
    }

    public List<UserResponseTo> getAll() {
        return userRepository.findAll().stream()
                .map(userMapper::toDto)
                .collect(Collectors.toList());
    }

    public UserResponseTo getById(Long id) {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new NotFoundException("User not found with id: " + id));
        return userMapper.toDto(user);
    }

    public UserResponseTo create(UserRequestTo request) {
        User user = userMapper.toEntity(request);
        User saved = userRepository.save(user);
        return userMapper.toDto(saved);
    }

    public UserResponseTo update(Long id, UserRequestTo request) {
        User user = userRepository.findById(id)
                .orElseThrow(() -> new NotFoundException("User not found with id: " + id));
        userMapper.updateEntity(user, request);
        user.setModified(LocalDateTime.now());
        User updated = userRepository.save(user);
        return userMapper.toDto(updated);
    }

    public void delete(Long id) {
        if (!userRepository.existsById(id)) {
            throw new NotFoundException("User not found with id: " + id);
        }
        userRepository.deleteById(id);
    }
}
